src/ConfigManager.h

```diff
@@ -0,0 +1,172 @@
+    static void SetNK40Decays(G4int value)                { Instance()->nK40Decays_ = value; }
+    static void SetNK42Decays(G4int value)                { Instance()->nK42Decays_ = value; }
+    static void SetNBi214Decays(G4int value)              { Instance()->nBi214Decays_ = value; }
+    static void SetNPb214Decays(G4int value)              { Instance()->nPb214Decays_ = value; }
+    static void SetNPo210Decays(G4int value)              { Instance()->nPo210Decays_ = value; }
+    static void SetNRn222Decays(G4int value)              { Instance()->nRn222Decays_ = value; }
+    static void SetEventCutoff(G4double value)            { Instance()->eventCutoff_ = value; }
+    static void SetEventWindow(G4double value)            { Instance()->eventWindow_ = value; }
+    static void SetSNTimingOn(G4bool value)               { Instance()->snTimingOn_ = value; }
+    static void SetTh2Name(G4String value)                { Instance()->th2Name_ = value; }            
+    static void SetDetectorConfiguration(G4bool value)    { Instance()->detectorConfiguration_ = value; }
+    static void SetDetectorLength(G4double value)         { Instance()->detectorLength_ = value; }
+    static void SetDetectorWidth(G4double value)          { Instance()->detectorWidth_ = value; }
+    static void SetDetectorHeight(G4double value)         { Instance()->detectorHeight_ = value; }
+
+    // Print out all configuration settings
+    static void Print() { Instance()->PrintConfig(); }
+    void PrintConfig() const;
+
+  private:
+    // All messengers
+    G4GenericMessenger* msgEvent_;
+    G4GenericMessenger* msgInputs_;
+    G4GenericMessenger* msgSupernova_;
+    G4GenericMessenger* msgSupernovaTiming_;
+    G4GenericMessenger* msgGeometry_;
+
+  private:
+    G4bool        decayAtTimeZero_;
+    G4bool        isotropic_;
+    G4bool        multirun_;
+    G4bool        overrideVertexPosition_;
+    G4bool        printParticleInfo_;
+    G4bool        snTimingOn_;
+
+    G4double      energyThreshold_;
+    G4double      eventCutoff_;
+    G4double      eventWindow_;
+    G4double      vertexX_;
+    G4double      vertexY_;
+    G4double      vertexZ_;
+
+    G4int         eventIDOffset_;
+    G4int         nAr39Decays_;
+    G4int         nAr42Decays_;
+    G4int         nBi214Decays_;
+    G4int         nCo60Decays_;
+    G4int         nK40Decays_;
+    G4int         nK42Decays_;
+    G4int         nKr85Decays_;
+    G4int         nPb214Decays_;
+    G4int         nPo210Decays_;
+    G4int         nRn222Decays_;
+
+    G4String      generator_;
+    G4String      genieFormat_;
+    G4String      inputFile_;    
+    G4String      marleyJson_;
+    G4String      outputFile_;
+    G4String      particleType_;
+    G4String      th2Name_;
+
+    G4bool        detectorConfiguration_;
+
+    G4ThreeVector momentumDirection_;
+    G4double      detectorLength_;
+    G4double      detectorWidth_;
+    G4double      detectorHeight_;
+};
+
+     
+
```

Comment:
Delete these extraneous blank lines.

Reply:
fixing in bugfix/v3.0.0_revisions branch